src/PIL/JpegImagePlugin.py

```diff
@@ -585,9 +585,11 @@ def _getmp(self):
 
 
 def convert_dict_qtables(qtables):
-    qtables = [qtables[key] for key in range(len(qtables)) if key in qtables]
-    for idx, table in enumerate(qtables):
-        qtables[idx] = [table[i] for i in zigzag_index]
+    warnings.warn(
+        "convert_dict_qtables is deprecated and will be removed in a future"
+        " release. Conversion is no longer needed.",
+        DeprecationWarning,
+    )
```

Comment:
A thought (feel free to disagree) - if we're considering this to be a public method that requires deprecation, then should we maybe just leave the functionality there (still adding in the deprecation warning)?

Reply:
I searched GitHub for this function: https://github.com/search?q=convert_dict_qtables+language%3Apython+-filename%3AJpegPresets.py. There are not many uses. Only the first two pages contain interesting results. The rest is just copies of Pillow itself. They all take the `.quantization` of an existing image, convert it to a nested list with this function and then manipulate that nested list in some way.

If Pillow de-zigzags the tables when loading (as the rest of this PR does), `.quantization` no longer needs any conversion step. That means those existing uses of the `convert_dict_qtables` function will actually break their code. The order of the numbers in the tables will be wrong. They'd need to remove the function call to fix their code. Turning `convert_dict_qtables` into a no-op means their code will continue to work unchanged. They'll only need to remove the function call to get rid of the warning.

Most people don't need to mess with quantization tables. But there's obviously more code than just what I found on GitHub and I don't know how representative these uses are. I guess there could also be cases where people are getting qtables from somewhere _other_ than straight from an image's `.quantization`. Maybe they serialized them for later re-use? In that case they no longer have a helper function that converts them back.

I guess Pillow could temporarily add a new `really_convert_dict_qtables` for those who still need it. :wink:

Reply:
@radarhere What do you think? Shall we merge this PR as is?

(And can explicitly replace "a future version" with "Pillow 10 (2023-01-02)")

Reply:
Sure - I can see now why `convert_dict_qtables` should do nothing, so no objections from me.